google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -143,17 +145,18 @@ public GapicSpannerRpc(SpannerOptions options) throws IOException {
             mergedHeaderProvider.getHeaders(),
             internalHeaderProviderBuilder.getResourceHeaderKey());
 
-    // TODO(pongad): make channel pool work
-
     // TODO(pongad): make RPC logging work (formerly LoggingInterceptor)
     // TODO(pongad): add watchdog
     // TODO(pongad): make error augmentation work (formerly SpannerErrorInterceptor)
 
     TransportChannelProvider channelProvider =
-        FixedTransportChannelProvider.create(
-            GrpcTransportChannel.newBuilder()
-                .setManagedChannel(options.getRpcChannels().get(0))
-                .build());
+        InstantiatingGrpcChannelProvider
+            .newBuilder()
+            .setEndpoint(options.getEndpoint())
+            .setMaxInboundMessageSize(MAX_MESSAGE_SIZE)
+            .setPoolSize(options.getNumChannels())
+            .build();
```

Comment:
We want to use the same channel for the same cloud spanner session. Does that work with this channel pool?

Reply:
Yes. As long as the `channelHint` is set in the `options` passed to rpc calls, the `channelPool` works the same way as before.

Reply:
Can we also set MAX_HEADER_LIST_SIZE ? 

causes this bug - https://b.corp.google.com/issues/65545193

Reply:
It seems io.grpc.ManagedChannelBuilder does not have MAX_HEADER_LIST_SIZE yet (https://grpc.io/grpc-java/javadoc/io/grpc/ManagedChannelBuilder.html#maxInboundMessageSize-int-)? @pongad can you confirm?

Reply:
FYI, MAX_HEADER is not optional. It has caused issues in the past and it will best to maintain parity when switching Spanner client to use the underlying library

Reply:
I'll add a TODO first and see if it would be possible to add this in gax and/or grpc